SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -137,38 +142,61 @@ void USpatialReceiver::OnAddComponent(Worker_AddComponentOp& Op)
 		return;
 	}
 
-	// If a client gains ownership over something it had already checked out, it will
-	// add component interest on the owner only data components, which will trigger an
-	// AddComponentOp, but it is not guaranteed to be inside a critical section.
-	if (!NetDriver->IsServer())
+	if (bInCriticalSection)
 	{
-		if (USpatialActorChannel* Channel = NetDriver->GetActorChannelByEntityId(Op.entity_id))
-		{
-			if (ClassInfoManager->GetCategoryByComponentId(Op.data.component_id) == SCHEMA_OwnerOnly)
-			{
-				// We received owner only data, and we have the entity checked out already,
-				// so this happened as a result of adding component interest. Apply the data
-				// immediately instead of queuing it up (since there will be no AddEntityOp).
-				ApplyComponentData(Op.entity_id, Op.data, Channel);
-				return;
-			}
-		}
+		PendingAddComponents.Emplace(Op.entity_id, Op.data.component_id, MakeUnique<DynamicComponent>(Op.data));
 	}
-
-	if (!bInCriticalSection)
+	else
 	{
-		UE_LOG(LogSpatialReceiver, Warning, TEXT("Received a dynamically added component, these are currently unsupported - component ID: %u entity ID: %lld"),
-			Op.data.component_id, Op.entity_id);
-		return;
+		HandleDynamicAddComponent(Op);
```

Comment:
This function has a precondition that the component is not in the view, and it looks like the component must always be in the view after. So we should probably move adding it to the view as part of this function.